compiler/core/lam.mli

```diff
@@ -74,7 +74,7 @@ and t = private
   | Lstringswitch of t * (string * t) list * t option
   | Lstaticraise of int * t list
   | Lstaticcatch of t * (int * ident list) * t
-  | Ltrywith of t * ident * t
+  | Ltrywith of t * ident * t option * t option
```

Comment:
opportunity: this file is entirely uncommented
a separate PR could *assist* in adding some explanation

More bigger / longer term assistance opportunity: why do we have both lambda and lam? Is that a historical artefact for what used to be backwards compatibility, or can they possibly be merged now?